spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+RSpec.describe OmniauthCallbacksController, :type => :controller do
+
+  describe "#facebook" do
+
+    context "given user does persist" do
+      before do
+        OmniAuth.config.mock_auth[:do_omniauth] = FactoryGirl.create(:auth_hash, :facebook)
+
+        request.env["devise.mapping"] = Devise.mappings[:user]
+        request.env["omniauth.auth"] = OmniAuth.config.mock_auth[:do_omniauth]
+      end
+
+      it "should successfully authenticate using Facebook authentication hash and display a success message in flash" do
+        get :facebook
+        user = User.find_by(email: 'testuser@facebook.com')
+        expect(user.id).not_to eq(nil)
```

Comment:
`user.id` can never be `nil`. Either the user will be in the DB and will have a valid ID, or the user won't be present, and `find_by` will return `nil`.